addons/mail/models/mail_mail.py

```diff
@@ -81,6 +85,36 @@ def default_get(self, fields):
     scheduled_date = fields.Char('Scheduled Send Date',
         help="If set, the queue manager will send the email after the date. If not set, the email will be send as soon as possible.")
 
+    @api.depends('attachment_ids')
+    def _compute_restricted_attachments(self):
+        """We might not have access to all the attachments of the emails.
+
+        Compute the attachments we have access to,
+        and the number of attachments we do not have access to.
+        """
+        for mail in self:
+            mail.unrestricted_attachment_ids = self._filter_attachment_access(mail.attachment_ids)
```

Comment:
Seems _filter_attachment_access could use mail as recordset, and use self.attachment_ids (therefore removing the need of a paraemter)